### PR TITLE
fix HostAliases' json keys to be hostAlias instead of hostMapping to reflect actual feature name

### PR DESCRIPTION
**Pull request by evie404**
**What this PR does / why we need it**: a rename was introduce during the middle of #44641 to change from `hostMappings` to `hostAliases`. the Go structs were updated, but I neglected to update the json keys. They should be in sync.

**Special notes for your reviewer**:  I messed up. This is an API change. I hope this is still ok to be in the 1.7 release.

**Release note**:
```release-note
HostAliases is now parsed with `hostAliases` json keys to be in line with the feature's name.
```

### CODE REVIEW

**Comment by rickypai** [comment]
@kubernetes/kubernetes-release-managers this is an API change

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
    "title": "Kubernetes",
-   "version": "v1.8.0"
+   "version": "v1.4.0"
```
Comment (cblecker) [inline api/openapi-spec/swagger.json]: Why is this changing? It shouldn't.

File: federation/apis/openapi-spec/swagger.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "swagger": "2.0",
   "info": {
    "title": "Generic API Server",
-   "version": "v1.8.0"
+   "version": "v1.4.0"
```
Comment (cblecker) [inline federation/apis/openapi-spec/swagger.json]: Same here

**Review by cblecker — CHANGES_REQUESTED** [review]

**Comment by nikhiljindal** [comment]
@k8s-bot ok to test.

Keeping actual changes and autogenerated changes in different commits makes it easier to review.
This is an API change so it will break existing users.

Will let Tim do the review.

File: pkg/api/v1/types.go [hunk 3]
```diff
@@ -2519,7 +2519,7 @@ type PodSpec struct {
 	// +optional
 	// +patchMergeKey=IP
 	// +patchStrategy=merge
-	HostAliases []HostAlias `json:"hostMappings,omitempty" patchStrategy:"merge" patchMergeKey:"IP" protobuf:"bytes,23,rep,name=hostMappings"`
+	HostAliases []HostAlias `json:"hostAliases,omitempty" patchStrategy:"merge" patchMergeKey:"IP" protobuf:"bytes,23,rep,name=hostAliases"`
```
Comment (liggitt) [inline pkg/api/v1/types.go]: should the patch merge key should be lowercase `ip` as well?

File: api/openapi-spec/swagger.json (hunk 1 above)
Comment (rickypai) [inline api/openapi-spec/swagger.json]: not sure. i ran this on a new development environment. will manually remove the change

File: federation/apis/openapi-spec/swagger.json (hunk 2 above)
Comment (rickypai) [inline federation/apis/openapi-spec/swagger.json]: fixed

File: pkg/api/v1/types.go (hunk 3 above)
Comment (rickypai) [inline pkg/api/v1/types.go]: thanks for catching this also!

**Comment by cblecker** [comment]
@k8s-bot ok to test

**Comment by dchen1107** [comment]
@erictune I am assigning this to you since @thockin is out. Please let me know you couldn't review this. Thanks!

cc/ @kubernetes/api-reviewers

**Review by erictune — APPROVED** [review]

**Comment by rickypai** [comment]
thank you!

**Comment by rickypai** [comment]
@erictune I think you need to explicitly /lgtm this PR or else it won't be picked up by submit queue.

thanks a lot!

**Comment by rickypai** [comment]
recent merge introduced conficts in the generated protobuf files. rebased and regenerating the files.

**Comment by caesarxuchao** [comment]
Thanks for the rebasing. lgtm.

**Comment by smarterclayton** [comment]
@caesarxuchao we should probably open an issue in 1.8 to track API changes and make sure they conform to conventions.

**Comment by caesarxuchao** [comment]
> we should probably open an issue in 1.8 to track API changes and make sure they conform to conventions.

@mbohlool i heard you had started an API convention checker?

**Comment by rickypai** [comment]
API convention checker would be nice. i was also thinking requiring e2e test for every API change to mitigate mistakes like this

**Comment by caesarxuchao** [comment]
Adding queue/blocks-others because it's blocking lifting code freeze.

**Comment by rickypai** [comment]
thanks and sorry that this is blocking folks :(

**Comment by rickypai** [comment]
yay!

**Comment by mbohlool** [comment]
> we should probably open an issue in 1.8 to track API changes and make sure they conform to conventions.
> @mbohlool i heard you had started an API convention checker?

#44005

**Pull request merged**